src/service/transaction.service.ts

```diff
@@ -50,178 +53,235 @@ import {
 } from 'nem2-sdk';
 
 export class TransactionService {
-
+    private table: HorizontalTable;
     constructor() {
 
     }
 
     public formatTransactionToFilter(transaction: Transaction): string {
         let transactionFormatted = '';
+        const propertyList: any[] = [];
         if (transaction instanceof TransferTransaction) {
-            transactionFormatted += 'TransferTransaction: RecipientAddress:';
+            this.table = new Table({
+                style: { head: ['cyan'] },
+                head: ['RecipientAddress', 'Message', 'Mosaics', 'SignerPublicKey', 'SignerPublicKey', 'Deadline', 'Hash'],
+            }) as HorizontalTable;
+
+            transactionFormatted += 'TransferTransaction:\n';
             if (transaction.recipientAddress instanceof Address) {
-                transactionFormatted += transaction.recipientAddress.pretty();
+                propertyList[0] = transaction.recipientAddress.pretty();
             } else {
-                transactionFormatted += transaction.recipientAddress.toHex();
+                propertyList[0] = transaction.recipientAddress.toHex();
             }
-            transactionFormatted += transaction.message.payload.length > 0 ? ' Message:\"' + transaction.message.payload + '\"' : '';
+            propertyList[1] = transaction.message.payload.length > 0 ? ' Message:\"' + transaction.message.payload + '\"' : '';
             if (transaction.mosaics.length > 0) {
-                transactionFormatted += ' Mosaics: ';
+                let mosaicText = '';
                 transaction.mosaics.map((mosaic) => {
                     if (mosaic.id instanceof MosaicId) {
-                        transactionFormatted += 'MosaicId:';
+                        mosaicText += 'MosaicId:';
                     } else {
-                        transactionFormatted += 'NamespaceId:';
+                        mosaicText += 'NamespaceId:';
                     }
-                    transactionFormatted += mosaic.id.toHex() + '::' + mosaic.amount.compact() + ',';
+                    mosaicText += mosaic.id.toHex() + '::' + mosaic.amount.compact() + ',';
                 });
-                transactionFormatted = transactionFormatted.substr(0, transactionFormatted.length - 1);
+                mosaicText = mosaicText.substr(0, mosaicText.length - 1);
+                propertyList[2] = mosaicText;
             }
         } else if (transaction instanceof NamespaceRegistrationTransaction) {
-            transactionFormatted += 'NamespaceRegistrationTransaction: NamespaceName:' + transaction.namespaceName;
+            this.table = new Table({
```

Comment:
Create one class per transaction table, so we can test each table in the future.

Example: https://github.com/nemtech/nem2-cli/blob/master/src/commands/block/header.ts#L36